lib/puppet/type/file/source.rb

```diff
@@ -250,6 +235,10 @@ def found?
     def copy_source_value(metadata_method)
       param_name = (metadata_method == :checksum) ? :content : metadata_method
       if resource[param_name].nil? or resource[param_name] == :absent
+        if Puppet.features.microsoft_windows? && [:owner, :group, :mode].include?(metadata_method)
+          devfail "Should not have tried to use source owner/mode/group on Windows"
+        end
+
```

Comment:
I'm not sure that was allowed. I think the failure above would've precluded it. I'll try it out.